nimx/naketools.nim

```diff
@@ -405,18 +405,39 @@ proc replaceVarsInFile(file: string, vars: Table[string, string]) =
         content = content.replace("$(" & k & ")", v)
     writeFile(file, content)
 
+proc chomp(s: string): string =
+    ## see https://dlang.org/library/std/string/chomp.html
+    if s.endsWith "\n":
+        result = s[0 ..< ^1]
+    else:
+        result = s
+
 proc buildSDLForDesktop(b: Builder): string =
     when defined(linux):
         result = "/usr/lib"
     elif defined(macosx):
-        if fileExists("/usr/local/lib/libSDL2.a") or fileExists("/usr/local/lib/libSDL2.dylib"):
-            result = "/usr/local/lib"
-        else:
-            let xcodeProjDir = expandTilde(b.sdlRoot)/"Xcode/SDL"
-            let libDir = xcodeProjDir/"build/Release"
-            if not fileExists libDir/"libSDL2.a":
-                direShell "xcodebuild", "-project", xcodeProjDir/"SDL.xcodeproj", "-target", "Static\\ Library", "-configuration", "Release", "-sdk", "macosx"&b.macOSSDKVersion, "SYMROOT=build"
-            result = libDir
+        proc isValid(dir: string): bool =
+            fileExists(dir / "libSDL2.a") or fileExists(dir / "libSDL2.dylib")
+        result = "/usr/local/lib"
+        if isValid(result): return result
+
+        if "brew --version".execCmdEx().exitCode == 0:
+            # user has homebrew
+            let ret = "brew --prefix sdl2".execCmdEx()
+            if ret.exitCode == 0:
+                result = ret.output.string.chomp / "lib"
```

Comment:
Why not `strutils.strip` instead of `chomp`?

Reply:
well that's different semantics so in general wouldn't be applicable (maybe in this particular case would work); 
* r u ok if i use strutils.stripLineEnd which was just merged here https://github.com/nim-lang/Nim/pull/9346 ? (would require nim >= 0.19.2 or whatever next point release is)
* if not, I'll work around that for this particular case

Reply:
> r u ok if i use strutils.stripLineEnd

sure.

Reply:
done